apps/project-renderer/src/components/app/index.js

```diff
@@ -129,14 +131,14 @@ const App = ( {
 			'crowdsignal-forms/text-question': TextQuestion,
 		} );
 
-	if ( hasResponded ) {
+	if ( hasResponded && ! hasErrors ) {
```

Comment:
`hasResponded` shouldn't even be set until you've successfully submitted your response - I think the additional condition shouldn't be necessary?

Reply:
Agree